src/PostgREST/RangeQuery.hs

```diff
@@ -38,12 +41,13 @@ rangeParse range = do
       rangeIntersection lower upper
     Nothing -> rangeGeq 0
 
-rangeRequested :: RequestHeaders -> NonnegRange
-rangeRequested = rangeParse . fromMaybe "" . lookup hRange
+rangeRequested :: RequestHeaders -> Maybe NonnegRange
```

Comment:
I don't particularly like this solution, it's like using (return -1) to signal Nothing in C for the sake of making the code a few lines shorter.